tests/impls/perturb_image/test_rise.py

```diff
@@ -0,0 +1,146 @@
+        """
+        impl1 = RISEPertubation(N=1000, s=8, p1=0.5)
+        impl2 = RISEPertubation(N=1000, s=8, p1=0.5)
+        assert not np.array_equal(impl1.grid, impl2.grid)
+
+    def test_seed(self) -> None:
+        """
+        Test that passing a seed generates equivalent masks
+        """
+        impl1 = RISEPertubation(N=1000, s=8, p1=0.5, seed=42)
+        impl2 = RISEPertubation(N=1000, s=8, p1=0.5, seed=42)
+        assert np.array_equal(impl1.grid, impl2.grid)
+
+    def test_perturb_1channel(self) -> None:
+        """
+        Test basic perturbation on a known image with even windowing + stride.
+        """
+        impl = RISEPertubation(N=2, s=2, p1=0.5, seed=42)
+        # Image is slightly wide, should be occluded 6-ways.
+        white_image = PIL.Image.fromarray(
+            np.full((4, 6), fill_value=255, dtype=np.uint8)
+        )
+        pert_imgs, pert_masks = impl.perturb(white_image)
+        assert len(pert_imgs) == 2
+        assert len(pert_masks) == 2
+        assert np.allclose(pert_masks, EXPECTED_MASKS_4x6)
+        # Output image modes should match input
+        for i, img in enumerate(pert_imgs):
+            assert img.mode == "L"
+            # Test for expected output perturbed image content
+            assert np.allclose(img, EXPECTED_IMAGES_4x6_L[i])
+
+    def test_perturb_3channel(self) -> None:
+        """
+        Test basic perturbation on a known image with even windowing + stride.
+        """
+        impl = RISEPertubation(N=2, s=2, p1=0.5, seed=42)
+        # Image is slightly wide, should be occluded 6-ways.
+        white_image = PIL.Image.fromarray(
+            np.full((4, 6, 3), fill_value=255, dtype=np.uint8)
+        )
+        pert_imgs, pert_masks = impl.perturb(white_image)
+        assert len(pert_imgs) == 2
+        assert len(pert_masks) == 2
+        assert np.allclose(pert_masks, EXPECTED_MASKS_4x6)
+        # Output image modes should match input
+        for i, img in enumerate(pert_imgs):
+            assert img.mode == "RGB"
+            # Test for expected output perturbed image content
+            expected_img_ch = EXPECTED_IMAGES_4x6_L[i]
+            expected_img = np.stack(
+                (expected_img_ch, expected_img_ch, expected_img_ch),
+                -1
+            )
+            assert np.allclose(img, expected_img)
+
+    def test_multiple_image_sizes(self) -> None:
+        """
+        Test that once we initialize a RISEPerturbation we can call it on
+        images of varying sizes
+        """
+        impl = RISEPertubation(N=2, s=2, p1=0.5, seed=42)
+        white_image_small = PIL.Image.fromarray(
+            np.full((4, 6), fill_value=255, dtype=np.uint8)
+        )
+        white_image_large = PIL.Image.fromarray(
+            np.full((41, 26), fill_value=255, dtype=np.uint8)
+        )
+        pert_imgs_small, pert_masks_small = impl.perturb(white_image_small)
+        assert pert_imgs_small[0].size == white_image_small.size
```

Comment:
A simple assert missing here is that of checking the quantity of masks being returned:
```suggestion
        pert_imgs_small, pert_masks_small = impl.perturb(white_image_small)
        assert len(pert_imgs_small) == len(pert_masks_small) == 2  # `N` value above.
        assert pert_imgs_small[0].size == white_image_small.size
```